### PR TITLE
Allow frontend to set system time

### LINKED ISSUE #74: Add a mechanism for syncing clock time with topside

### PR DESCRIPTION
**Pull request by patrickelectric**
Fix #74 

docker: patrickelectric/blueos-core:set_time

### CODE REVIEW

File: core/services/commander/main.py [hunk 1]
```diff
@@ -87,6 +88,18 @@ async def command_host(response: Response, command: str, i_know_what_i_am_doing:
         return {"message": f"{error}"}
 
 
+@app.post("/set_time", status_code=status.HTTP_200_OK)
+@version(1, 0)
+async def set_time(response: Response, unix_time_seconds: int, i_know_what_i_am_doing: bool = False) -> Any:
+    unix_time_seconds_now = int(time.time())
+    if abs(unix_time_seconds_now - unix_time_seconds) < 30 * 60:
```
Comment (Williangalvani) [inline core/services/commander/main.py:95]: can you think of any reasons not to have a lower number here? 30 minutes is enough time to turn off the rov and move it somewhere else, which would result in some logs with the wrong time. not a great issue, but slightly inconvenient

Comment (patrickelectric) [reply core/services/commander/main.py:95]: Any time may be arbitrary, what you suggest ? 5min ?

File: core/services/commander/main.py [hunk 2]
```diff
@@ -87,6 +88,18 @@ async def command_host(response: Response, command: str, i_know_what_i_am_doing:
...
+        return {"message": f"External time ({unix_time_seconds}) is close to internal time ({unix_time_seconds_now})."}
```
Comment (Williangalvani) [inline core/services/commander/main.py:96]: add a ", not updating." in the end.

File: core/services/commander/main.py (hunk 1 above)
Comment (patrickelectric) [reply core/services/commander/main.py]: Updated to 5min.

**Review by Williangalvani — COMMENTED** [review]
Looks good. any idea if we should be worried about timezones?
or is everything utc?

any tips to test it?

**Comment by patrickelectric** [comment]
> Looks good. any idea if we should be worried about timezones? or is everything utc?
> 
> any tips to test it?

`getTime` return time in `unix time format`, that is [by default UTC](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#the_ecmascript_epoch_and_timestamps). the _date command_ accepts unix time with the `@`, that is also UTC.

File: core/services/commander/main.py (hunk 2 above)
Comment (patrickelectric) [reply core/services/commander/main.py]: Done.

**Review by Williangalvani — APPROVED** [review]

**Pull request merged**